src/LeafMutator.ts

```diff
@@ -0,0 +1,47 @@
+import ActionCreator from './interfaces/ActionCreator';
+import ActionMessage from './interfaces/ActionMessage';
+import { getPrivateActionId } from './actionCreator';
+
+export type MutatorHandler<TState, TAction extends ActionMessage> = (
+    state: TState,
+    action: TAction
+) => TState | void;
+
+// Represents a mutator for a leaf node in the state tree
+export default class LeafMutator<TState> {
+    private handlers: { [actionId: string]: MutatorHandler<TState, ActionMessage> } = {};
+
+    constructor(private initialValue: TState) {}
+
+    getInitialValue() {
+        return this.initialValue;
```

Comment:
Is it a concern that if a mutable value is passed, the `initialValue` could change over time?

Reply:
This is a good point.  No, I wasn't worried about this because MobX creates a clone of the value rather than allowing the original value to be modified.  However, this is only true of more recent versions of MobX.  So this could be busted in the v3 branch of satchel because it allows use of MobX 2.  That's a bummer, because I had intended to cherry-pick this back into v3.  I'm not sure if this is a big enough gotcha to avoid doing that or not.

---
In reply to: [200790349](https://github.com/Microsoft/satcheljs/pull/100#discussion_r200790349) [](ancestors = 200790349)